packages/adp-tooling/src/preview/routes-handler.ts

```diff
@@ -170,6 +171,65 @@ export default class RoutesHandler {
         }
     };
 
+    /**
+     * Handler for retrieving existing controller extension data from the workspace.
+     *
+     * @param req Request
+     * @param res Response
+     * @param next Next Function
+     */
+    public handleGetControllerExtensionData = async (req: Request, res: Response, next: NextFunction) => {
+        try {
+            const params = req.params as { controllerName: string };
+            const controllerName = sanitize(params.controllerName);
+            const codeExtFiles = await this.readAllFilesByGlob('/**/changes/*_codeExt.change');
+
+            let controllerExists = false;
+            let controllerPath = '';
+            let controllerPathFromRoot = '';
+
+            const project = this.util.getProject();
+            const sourcePath = project.getSourcePath();
+            const projectName = project.getName();
+
+            const getPath = (projectPath: string, fileName: string) =>
+                path.normalize(path.join(projectPath, FolderNames.Changes, FolderNames.Coding, fileName));
+
+            for (const file of codeExtFiles) {
+                const fileStr = await file.getString();
+                const change = JSON.parse(fileStr) as CodeExtChange;
+
+                if (change.selector.controllerName === controllerName) {
+                    const fileName = change.content.codeRef.replace('coding/', '');
+                    controllerPath = getPath(sourcePath, fileName);
+                    controllerPathFromRoot = getPath(projectName, fileName);
+                    controllerExists = true;
+
+                    if (!fs.existsSync(controllerPath)) {
+                        const errorMsg = `Controller extension file was not found at ${controllerPath}`;
+                        this.logger.debug(errorMsg);
+                        throw new Error(errorMsg);
```

Comment:
Could you please replace `throw new Error(errorMsg)` with your previous implementation:
```
                res.status(HttpStatusCodes.NOT_FOUND).send({
                    message: errorMsg
                });
```
Throwing an error is too hard. Sorry for asking to change again.

Reply:
Changed in: [65646e7](https://github.com/SAP/open-ux-tools/pull/1408/commits/65646e72c170a4b0b2809ba12689bc4291c85c89)

Reply:
Thanks